lib/instrumentation/http-shared.js

```diff
@@ -191,13 +189,11 @@ exports.traceOutgoingRequest = function (agent, moduleName, method) {
         return emit.apply(req, arguments)
       }
 
-      let url, statusCode
-      httpRequestToUrl(req).then(_url => {
-        url = _url
-      }).catch(() => {
+      const url = getUrlFromRequestAndOptions(req, options, moduleName)
```

Comment:
```suggestion
      const url = getUrlFromRequestAndOptions(req, options, moduleName + ':')
```

Need the colon, otherwise you get something like: `http//localhost/get` in the *very* special circumstances where `fallbackProtocol` is used. E.g. with node 10 or less, using this:

```
http.get(
  {
    hostname: 'localhost',
    path: '/get',
    // A custom agent that implements the minimum to pass muster, but does
    // *not* define `agent.protocol`.
    agent: {
      addRequest() {}
    }
  }
)
```

Reply:
got it -- I've made this change, and added the test I should have to test this particular case.